core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -184,6 +184,34 @@ class GroupMetadataManager(brokerId: Int,
     }
   }
 
+  /**
+    * Delete groups
+    */
+  def deleteGroups(groupIds: Seq[String]): Map[String, Errors] = {
+    var eligibleGroups: Seq[GroupMetadata] = Seq()
+    var result: Map[String, Errors] = Map()
+
+    groupIds.foreach { groupId =>
+      if (!groupMetadataCache.contains(groupId))
```

Comment:
This "check and act" is not safe since we're not holding a lock. It would be better to `get` the `GroupMetadata` object and check if it is null. If it is not null, then we need to grab the group lock before checking its state and attempting to delete its state.